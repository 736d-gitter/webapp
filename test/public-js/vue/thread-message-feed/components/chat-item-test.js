'use strict';

const mount = require('../../vuex-mount');
const {
  default: ChatItem
} = require('../../../../../public/js/vue/thread-message-feed/components/chat-item.vue');

describe('thread-message-feed chat-item', () => {
  const defaultProps = {
    message: {
      id: '5d147ea84dad9dfbc522317a',
      text: 'Example message using a bit of  `code` and **bold** to show how *markdown* is stored.',
      html:
        'Example message using a bit of  <code>code</code> and <strong>bold</strong> to show how <em>markdown</em> is stored.',
      sent: '2019-06-27T08:30:32.165Z',
      fromUser: {
        id: '5cdc09f6572f607a5bc8a41d',
        username: 'viktomas_gitlab',
        displayName: 'Tomas Vik',
        url: '/viktomas_gitlab',
        avatarUrl: 'http://localhost:5000/api/private/avatars/g/u/viktomas_gitlab',
        avatarUrlSmall:
          'https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=60&d=identicon',
        avatarUrlMedium:
          'https://secure.gravatar.com/avatar/6042a9152ada74d9fb6a0cdce895337e?s=128&d=identicon',
        v: 29
      }
    }
  };

  it('matches snapshot', () => {
    const { wrapper } = mount(ChatItem, defaultProps);
    expect(wrapper.element).toMatchSnapshot();
  });
});
