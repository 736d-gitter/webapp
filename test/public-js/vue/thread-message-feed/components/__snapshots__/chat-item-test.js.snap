// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed chat-item matches snapshot 1`] = `
<div
  class="chat-item burstStart thread-message-feed"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <!---->
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from js-chat-item-from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username js-chat-item-from"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time js-chat-time"
          title="June 27, 2019 10:30 AM"
        >
          Jun 27 10:30
        </a>
      </div>
       
      <div
        class="chat-item__text js-chat-item-text"
      >
        Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;
